framework/py/flwr/common/inflatable.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Inflatable base class."""
+
+
+import hashlib
+from typing import TypeVar
+
+from .constant import HEAD_BODY_DIVIDER
+
+T = TypeVar("T", bound="Inflatable")
+
+
+class Inflatable:
+    """Base class for inflatable objects."""
+
+    def deflate(self) -> tuple[bytes, str]:
+        """Deflate object."""
+        raise NotImplementedError()
+
+    @property
+    def object_id(self):
+        """Get object_id."""
+        _, object_id = self.deflate()
+        return object_id
+
+    @object_id.setter
+    def object_id(self, value: str) -> None:
+        self._object_id = value
```

Comment:
Why does `object_id` have a setter? Shouldn't it only be set by internal methods, never by the user?